docs/toolhive/concepts/backend-auth.mdx

```diff
@@ -0,0 +1,285 @@
+
+### Same IdP with token exchange
+
+When both the MCP server and the backend service trust the same IdP, and that
+IdP supports [RFC 8693](https://datatracker.ietf.org/doc/html/rfc8693) token
+exchange, ToolHive can exchange the internal token for an external one.
+
+```mermaid
+flowchart LR
+    User[User]
+    IDP[Identity Provider]
+    MCP[MCP Server]
+    Upstream[Upstream Service]
+
+    User -->|login & receive token| IDP
+    MCP -->|subject_token → exchange| IDP
+    IDP -->|external_token issued| MCP
+    MCP -->|calls with external_token| Upstream
+    Upstream -->|validates token| IDP
+```
+
+**How it works:**
+
+1. The user authenticates to the MCP client and receives an access token from
+   the IdP
+2. ToolHive's token exchange middleware contacts the IdP, presenting the user's
+   access token
+3. The IdP issues a new access token with different audience and scopes
+4. ToolHive passes this access token to the MCP server
+5. The MCP server uses the access token to call the upstream service
+
+### Federated IdPs with identity mapping
+
+When the backend service trusts a different IdP, but federation is configured
+between the two IdPs, ToolHive can use the federated identity service to issue
+short-lived tokens. An example is Google's Security Token Service (STS), which
+can issue tokens for Google Cloud services based on your corporate identity.
+
+```mermaid
+flowchart LR
+    User[User]
+    IDP1[IdP A - User Login]
+    IDP2[IdP B - Federated]
+    MCP[MCP Server]
+    Upstream[Upstream Service]
+
+    User -->|login & receive token_A| IDP1
+    IDP1 -->|submit token_A| IDP2
+    IDP2 -->|issue token_B| IDP1
+    IDP1 -->|pass token_B| MCP
+    MCP -->|call with token_B| Upstream
+    Upstream -->|validate token_B| IDP2
+```
+
+**How it works:**
+
+1. The user authenticates to their MCP client with a corporate IdP and receives
+   token_A
+2. ToolHive submits token_A to the federated identity service
+3. The federated service maps the identity and issues token_B
+4. ToolHive passes token_B to the MCP server
+5. The MCP server uses token_B to call the upstream service
+
+## Token exchange implementation
+
+The token exchange flow demonstrates how ToolHive transforms user identity
+tokens into properly scoped service tokens.
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant ToolHive
+    participant IDP
+    participant MCP Server
+    participant Upstream Service
+
+    Client->>ToolHive: Request with user token
+    ToolHive->>IDP: Validate token
+    IDP-->>ToolHive: Token valid
+    ToolHive->>IDP: Exchange token request
+    IDP-->>ToolHive: Service token
+    ToolHive->>MCP Server: Request with service token
+    MCP Server->>Upstream Service: API call
+    Upstream Service-->>MCP Server: Response
+    MCP Server-->>ToolHive: Response
+    ToolHive-->>Client: Response
+```
+
+### Token transformation
+
+When a client authenticates to ToolHive, they receive a token scoped for the MCP
```

Comment:
it receives (client is singular)

Reply:
Fixed in 46c02b3. Changed "they receive" to "it receives" to match the singular "client". Good catch!